Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs

```diff
@@ -489,50 +488,55 @@ private void OnDestroy()
         /// </summary>
         /// <param name="type">The interface type for the system to be managed.  E.G. InputSystem, BoundarySystem</param>
         /// <param name="service">The Instance of the service class to register</param>
-        public void RegisterService(Type type, IMixedRealityService service)
+        public bool RegisterService(Type type, IMixedRealityService service)
         {
             if (ActiveProfile == null)
             {
                 Debug.LogError($"Unable to add a new {type.Name} Service as the Mixed Reality Toolkit has to Active Profile");
+                return false;
             }
 
             if (type == null)
             {
                 Debug.LogWarning("Unable to add a manager of type null.");
-                return;
+                return false;
             }
             if (service == null)
             {
                 Debug.LogWarning("Unable to add a manager with a null instance.");
-                return;
+                return false;
             }
 
             if (IsCoreSystem(type))
             {
                 IMixedRealityService preExistingService;
-                if (IsCoreSystem(type))
-                {
-                    ActiveProfile.ActiveServices.TryGetValue(type, out preExistingService);
-                }
-                else
-                {
-                    GetService(type, out preExistingService);
-                }
+
+                ActiveProfile.ActiveServices.TryGetValue(type, out preExistingService);
 
                 if (preExistingService == null)
                 {
                     ActiveProfile.ActiveServices.Add(type, service);
+                    return true;
                 }
                 else
                 {
                     Debug.LogError($"There's already a {type.Name} registered.");
+                    return false;
                 }
             }
             else
             {
-                MixedRealityComponents.Add(new Tuple<Type, IMixedRealityExtensionService>(type, (IMixedRealityExtensionService)service));
-                if (!isInitializing) { service.Initialize(); }
-                mixedRealityComponentsCount = MixedRealityComponents.Count;
+                try
```

Comment:
Instead of trying the catch the exception, let's do a check if it is assignable from `IMixedRealityExtensionService` and return false if not.

```
                if (IMixedRealityExtensionService.IsAssignableFrom(service))
                {
                    Debug.LogError($"Unable to register {service}. Concrete type is missing IMixedRealityExtensionService implementation.");
                    return;
                }
```